frontend/catalyst/pennylane_extensions.py

```diff
@@ -122,10 +122,13 @@ def __init__(self, fn, device):
 
     def __call__(self, *args, **kwargs):
         if isinstance(self, qml.QNode):
-            if self.device.short_name not in QFunc.RUNTIME_DEVICES:
+            if isinstance(self.device, qml.Device):
+                name = self.device.short_name
+            else:
+                name = self.device.name
```

Comment:
Is this to support both device API versions?

Reply:
Yes correct, `self.device.short_name` is from the old api and `self.device.name`  from the new one